### PR TITLE
Update to v2 of api

### PR DESCRIPTION
**Pull request by DanWilliamsAu**
This tool currently users v1 of the Smartling translation API.  This PR updates the Smartling CLI tool to use [v2 of the Smartling API](https://developer.smartling.com/v2.0/reference) 

# Approach

The approach that I took with this upgrade was to 'get it working' on the new API.  

It currently does not use any of the new features of the API.  There are some interesting features, such as the [strings API](https://developer.smartling.com/v2.0/reference#create-strings) and the [jobs API](https://developer.smartling.com/v2.0/reference#job) and the [get translations endpoint](https://developer.smartling.com/v2.0/reference#post_projects-projectid-locales-localeid-file-get-translations)

Once we have this merged, it would be good to spend some time thinking about how these new features might help with our translations workflow.

# Smartling golang SDK

I have opted to use [Smartling's official golang SDK](https://github.com/Smartling/api-sdk-go) in the CLI tool.  The majority of endpoints we required are implemented there, as well as authenticating and more.  Using the official SDK gets us closer to Smartling itself as well as prevents us from having to re-invent the wheel.

This means this PR removes our implementation of the API v1 Smartling client and library in favour of using the official SDK.  

A follow up item will be to use the official SDK in promotions https://github.com/99designs/promotions/blob/master/Gopkg.toml#L20

Unfortunately, the official SDK is not feature complete - I've had to add some endpoints to the SDK, I plan to get those new features merged into the official SDK (https://github.com/Smartling/api-sdk-go/pull/4)

# Why not just use smartling's official CLI tool?

Smartling has a [cli tool](https://github.com/Smartling/smartling-cli), which is a fork of this library.  Unfortunately that CLI tool is missing some features which means it won't work out of the box for our purposes.  The missing features include:
* a fault tolerant client that can work with smartling's regular errors such as "RESOURCE_LOCKED"
* automatic prefixing with git branch
* caching

### CODE REVIEW

File: cli/smartling/main.go [hunk 1]
```diff
@@ -44,31 +43,27 @@ var cmdBefore = func(c *cli.Context) error {
...
 		log.Fatalln("ApiKey not specified in --apikey or", configFile)
 	}
-	if projectId == "" {
-		log.Fatalln("ProjectId not specified in --projectid or", configFile)
+	if projectID == "" {
+		log.Fatalln("ProjectID not specified in --projectid or", configFile)
 	}
-
-	var sc *smartling.Client
-	if c.Bool("sandbox") {
```
Comment (sinamt) [inline cli/smartling/main.go:60]: We don't need sandbox support for anything?

File: cli/smartling/main.go [hunk 2]
```diff
@@ -44,31 +43,27 @@ var cmdBefore = func(c *cli.Context) error {
...
-		log.Println("Using sandbox")
-		sc = smartling.NewSandboxClient(apiKey, projectId)
-	} else {
-		sc = smartling.NewClient(apiKey, projectId)
+	if userID == "" {
+		log.Fatalln("UserID not specified in --userid or", configFile)
 	}
 
-	if timeout != 0 {
-		sc.SetHttpTimeout(time.Duration(timeout) * time.Second)
```
Comment (sinamt) [inline cli/smartling/main.go]: Is a timeout still supported somewhere?

**Review by sinamt — APPROVED** [review]
Nice work, looks pretty solid to me.

File: cli/smartling/main.go (hunk 1 above)
Comment (DanWilliamsAu) [reply cli/smartling/main.go:60]: Sandbox was an api v1 thing. There is no equivalent sandbox mode for API v2.

File: cli/smartling/main.go (hunk 2 above)
Comment (DanWilliamsAu) [reply cli/smartling/main.go]: Good question - i had removed that, but there was no reason to do so. I've added the timeout setting functionality back.

File: cli/smartling/api.go [hunk 3]
```diff
@@ -166,17 +141,16 @@ var PutCommand = cli.Command{
...
 	Action: func(c *cli.Context) {
-		if len(c.Args()) != 2 {
+		if len(c.Args()) != 1 {
 			log.Println("Wrong number of arguments")
-			log.Fatalln("Usage: put <local file> <remote file>")
+			log.Fatalln("Usage: put <local file>")
 		}
 
 		localpath := c.Args().Get(0)
-		remotepath := c.Args().Get(1)
```
Comment (mtibben) [inline cli/smartling/api.go:175]: Hmm how are we getting away with not having a remote path? Seems weird to me not specifying the remote path, how do we know where the file is going?

Comment (DanWilliamsAu) [reply cli/smartling/api.go:175]: The API v2 does not allow you to specify a remote path: https://developer.smartling.com/v2.0/reference#post_projects-projectid-file

You just provide the fileUri (in this case the localpath) and you can then use that to request the file back.

Comment (mtibben) [reply cli/smartling/api.go:175]: looks like two different fields to me?
```
file = The file contents to upload
fileUri = Value that uniquely identifies the uploaded file
```

Comment (DanWilliamsAu) [reply cli/smartling/api.go:175]: `file` is the actual _contents_ of the file, it does not having anything to do with the path of the file or identifying the file.  You can see how it is used here:  https://github.com/99designs/smartling/blob/master/api.go#L171

Comment (mtibben) [reply cli/smartling/api.go:175]: Yeah exactly, the local file is just the content, so why remove the remotepath part? I'm not sure the localpath = remotepath is a good assumption. Some examples of other unix cli tools always use `<source> <target>` arguments

```
$ cp
usage: cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file target_file
       cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file ... target_directory
```

```
$ scp
usage: scp [-346BCpqrv] [-c cipher] [-F ssh_config] [-i identity_file]
           [-l limit] [-o ssh_option] [-P port] [-S program]
           [[user@]host1:]file1 ... [[user@]host2:]file2
```

```
sftp put [-afPpr] local-path [remote-path]
```

Comment (DanWilliamsAu) [reply cli/smartling/api.go:175]: ahhh ok I see what you mean now, the penny has finally dropped! 

I'll patch that up.

**Pull request merged**